shelley/chain-and-ledger/dependencies/non-integer/doc/non-integer-calculations.tex

```diff
@@ -75,124 +85,163 @@
+
+An alternative approach is to use fixed precision\khcomment{Changed Fixed Point to Fixed Precision} arithmetic based on exact integers
+where a proportion of the integer is used as fractional part.
+The number of fractional digits is fixed to a constant.
+This is more efficient
+than using the rational number approach described above, since it is not necessary to normalise
+values after calculations, arbitrary precision results are not supported, and only a single
+arbitrarily sized integer needs to be stored.
+Since it is implemented purely in software on top of exact integer arithmetic, this approach
+will be less efficient than IEEE~754 floating point.
+However, it can be made to behave equivalently in different implementations.\khcomment{This is not clear: performance, accuracy, ...}
+%
 The basic idea is the following: for $n$ decimal digits of precision, multiply
-each integral value with $10^{n}$ and take this into account in the basic
-operations. To use in the case of Cardano, this would mean using $n=17$ to
-support the required 17 digits of precision to track each possible stake
-fraction.
+each integral value by $10^{n}$ and take this into account in the basic
+operations. For Cardano, this would mean using $n=17$ to
+support the required 17 digits of precision that would track each possible stake
+fraction.  Haskell provides library support for fixed precision numbers and arithmetic.
+
+\subsection{Exact Real Numbers}
+
+Finally, there has been some research work on exact representations of real
+numbers using e.g. \emph{continued fractions}, exploiting lazy evaluation to
+produce the required number of digits of precision\khcomment{Citation to
+  e.g. Martin Escardo's work}. All real numbers can be represented, whether rational or irrational.
+These approaches have not yet been widely used, however,
+except for research purposes.  Implementations are often complicated and
+expensive, and some algorithms may need to be changed (for example, exact equality may
+be impossible to determine in a finite time).
 
 \subsection{Conclusion}
 \label{sec:summary}
 
-For the Cardano cryptocurrency, the approach based on fixed point arithmetic
-seems to be the best adapted one. It allows for the desired precision and can be
+For the Cardano cryptocurrency, the fixed precision arithmetic approach
+seems to be most suitable. It allows for the desired precision and can be
 implemented in an equivalent way, because it uses exact integer
 arithmetic.~\footnote{There already exist two implementations, Haskell and C.}
-While the fixed-point approach is less efficient than IEEE~754, using partial
-pre-computation will be possible to speed it up in the case where the base of an
-exponentiation is a constant.
+While the fixed-precision approach is less efficient than IEEE~754 floating point,
+by using partial pre-computation it will be possible to improve performance in
+some cases (e.g. when using a constant base for exponentiation).
```

Comment:
maybe a bit more positive "improve performance considerably in many cases ... "

Reply:
👍